### PR TITLE
Use std::map for main dictionary in ensemble_config

### PR DESCRIPTION
**Pull request by joakim-hove**
There are many `hash_type` references left in the ensemble_config.cpp file - but this is a start.

### CODE REVIEW

File: lib/enkf/ensemble_config.cpp
Comment (andreabrambilla) [inline lib/enkf/ensemble_config.cpp]: in these situations where you have to check whether an object is present, and in that case do operations on it, it is more efficient to do it as follows
```
const auto node_it = ensemble_config->config_nodes.find(ecl_kw_name);
if(node_it != ensemble_config->config_nodes.end()) {
    const enkf_config_node_type * node = node_it.second;
    ...
```

File: lib/enkf/ensemble_config.cpp
Comment (andreabrambilla) [inline lib/enkf/ensemble_config.cpp]: same

File: lib/enkf/ensemble_config.cpp
Comment (andreabrambilla) [inline lib/enkf/ensemble_config.cpp]: same

File: lib/enkf/ensemble_config.cpp
Comment (andreabrambilla) [inline lib/enkf/ensemble_config.cpp:246]: this is slightly different but same concept

File: lib/enkf/ensemble_config.cpp
Comment (andreabrambilla) [inline lib/enkf/ensemble_config.cpp]: and this is the last one

**Review by andreabrambilla — APPROVED** [review]
Overall the PR looks good :+1:

I have added a comment that affects several places. Some could argue that's good practice, other can say it's a minor details given that performance is probably not critical here. Change it if you'd like too, merge it as is if you have better things to do :wink:

**Comment by joakim-hove** [comment]
Thank you for the review - I have updated according to your suggestion.

**Comment by joakim-hove** [comment]
> merge it as is if you have better things to do

Thank - I would if I could; but someone else must push the green ...

**Comment by joakim-hove** [comment]
Observe that I can not merge this myself - cool if someone with the magical power will push the green. The same applies the TestArea pr.

File: lib/enkf/ensemble_config.cpp
Comment (markusdregi) [inline lib/enkf/ensemble_config.cpp]: Does not seem like you use the key for anything?

**Comment by markusdregi** [comment]
I had one comment, besides that I’m all good for merging this one!

File: lib/enkf/ensemble_config.cpp
Comment (joakim-hove) [reply lib/enkf/ensemble_config.cpp]: Looks like you are right - fixed.

**Comment by markusdregi** [comment]
Good work!

**Pull request merged**
